### ISSUE TITLE
failed to trigger alert when instance is not up

### ISSUE THREAD

**Issue by jinjiaw** [VictoriaMetrics/VictoriaMetrics#967]
**Describe the bug**
It would be great [upgrading](https://victoriametrics.github.io/#how-to-upgrade) to [the latest avaialble release](https://github.com/VictoriaMetrics/VictoriaMetrics/releases)
and verifying whether the bug is reproducible there.
It is also recommended reading [troubleshooting docs](https://victoriametrics.github.io/#troubleshooting).

I am seeing that the alert was failed to be triggered when netdata on the target host is somehow stopped. 
we have following alter rule to evaluate the up metrics for netdata instances. The same rule worked well when we were using Prometheus. 
  - alert: 'Process netdata'
    expr: up{instance=~".*:19999",role!~"compute|node"} != 1
    for: 1m
    labels:
      serverity: critical
    annotations:
      summary: "Netdata on {{ $labels.instance }} is not running"
      description: "Netdata on {{ $labels.instance }} is not running"

Checking the targets via VictoriaMetrics API, we got the following regarding this failed node. The state is showing up while the last scrape is 40+ hours ago. 
state=up, endpoint=http://1.2.3.4:19999/api/v1/allmetrics?format=prometheus, labels={dc="location",instance="1.2.3.4:19999",job="server",prometheus="4.3.2.1",role="master",svc="k8s"}, last_scrape=170879.994s ago, scrape_duration=0.025s, error=""

**To Reproduce**
Steps to reproduce the behavior.

**Expected behavior**

we expected alert can be triggered. 

**Screenshots**

**Version**
The line returned when passing `--version` command line flag to binary. For example:
```
$ ./victoria-metrics-prod --version
victoria-metrics-20190730-121249-heads-single-node-0-g671d9e55
```

# ./victoria-metrics-prod --version
victoria-metrics-20201116-190828-tags-v1.47.0-0-g1c477bc2f

**Used command-line flags**
Command-line flags are listed as `flag{name="httpListenAddr", value=":443"} 1` lines at `/metrics` page.
See the following docs for details:

* [monitoring for single-node VictoriaMetrics](https://github.com/VictoriaMetrics/VictoriaMetrics/blob/master/README.md#monitoring)
* [montioring for VictoriaMetrics cluster](https://github.com/VictoriaMetrics/VictoriaMetrics/blob/cluster/README.md#monitoring)

**Additional context**
Add any other context about the problem here such as error logs from VictoriaMetrics and Prometheus,
`/metrics` output, screenshots from the official Grafana dashboards for VictoriaMetrics:

* [Grafana dashboard for single-node VictoriaMetrics](https://grafana.com/dashboards/10229)
* [Grafana dashboard for VictoriaMetrics cluster](https://grafana.com/grafana/dashboards/11176)

**Comment by valyala** [comment]
@jinjiaw , thanks for reporting the issue. It must be fixed in the commit b730fc266743bfdd94846ee2043cbcd1c4b6dd20 . It triggers when stream parsing is enabled with either `-promscrape.streamParse` command-line flag or `stream_parse: true` per-target config option.

**Comment by jinjiaw** [comment]
@valyala thanks much for your prompt action. As you mentioned, we did have -promscrape.streamParse flag.

**Comment by valyala** [comment]
The bugfix has been included in [v1.50.0](https://github.com/VictoriaMetrics/VictoriaMetrics/releases). Closing the issue as fixed.
